testsuite/tests/typing-gadts/test.ml

```diff
@@ -1073,6 +1073,10 @@ Error: The value "x" has type "t" = "< foo : int; .. as $0 >"
        but an expression was expected of type "< foo : int >"
        Type "$0" = "< bar : int; .. as $1 >" is not compatible with type "<  >"
        The second object type has no method "bar"
+       Hint: "$1" is a type variable introduced by the equation
+         "$0" = "< bar : int; .. as $1 >"
+       Hint: "$0" is a type variable introduced by the equation
+         "t" = "< foo : int; .. as $0 >"
```

Comment:
Here the hints are fully redundant with the earlier trace expansions. Later on, we might want to elide them.